fusedrug/utils/samplers/subset_sampler.py

```diff
@@ -0,0 +1,68 @@
+import torch
+import numpy as np
+from fuse.data.datasets.dataset_base import DatasetBase
+from typing import Optional, Union
+
+
+class SubsetSampler(torch.utils.data.Sampler):
```

Comment:
Is this similar to sub_epoch_sampler we have in bmfmcore?

Reply:
I think that it's a similar concept. this was done before I was aware of the bmfm one/perhaps before it existed